packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/query/ScalarQuery.kt

```diff
@@ -191,7 +222,19 @@ internal class SumQuery<E : BaseRealmObject, T : Any> constructor(
     }
 
     private fun findFromResults(resultsPointer: RealmResultsPointer): T = try {
-        computeAggregatedValue(resultsPointer, getPropertyKey(property))
+        getterScope {
+            val propertyKey = getPropertyKey(getFieldName())
+            val transport = realm_results_sum(resultsPointer, propertyKey)
+
+            when (type) {
+                // RealmAny SUMs are computed as Decimal128
+                RealmAny::class -> {
+                    val converter = primitiveTypeConverters[Decimal128::class]!!
+                    converter.realmValueToPublic(transport)
+                }
+                else -> coerceType(propertyMetadata, type, transport)
```

Comment:
`propertyMetadata` and `type` is known at query construction time, so we should have to reiterate the decisions on those on each `findFromResults`.